### DISCUSSION TITLE
example not working - He gas 1d

### DISCUSSION THREAD

**Discussion by elad2718** [netket/netket#1400] [Q&A] [answered]
The [He example](https://github.com/netket/netket/blob/master/Examples/Continuous/Helium.py) (as is given in the code and on the paper) does not converge at all to experimental value. On a good run, `65+-3` is achieved (instead of `~53`). This happens when I run with the exact code from the repo on colab (reproduced on local device).
The paper introduces a quick convergence that I cannot reproduce. Tried changing some parameters and still can't get convergence.

Also, each iteration takes about 4s (or 8 without gpu), that seems very slow to me for 4000 samples with a relatively small model.

**Comment by PhilipVinc** [comment]
@gpescia can you please look into this?

**Reply by PhilipVinc** [reply]
I just tried, and it seems that in the paper @gpescia used `nk.optimizer.Sgd(0.001)` while in the examples included in the repository he is using `nk.optimizer.Sgd(0.01)`.

running the two simulations for ~hundred steps, however does not seem to help reliably...
Maybe tomorrow @gpescia will elighten us
```python
# learning rate 0.01
➜ mpirun -np 10 ipython Helium.py
 15%|█▌        | 150/1000 [01:08<06:31,  2.17it/s, Energy=65.76 ± 0.71 [σ²=1443.00, R̂=1.0053, τ=1.4<4.4]]

# learning rate 0.001
➜ mpirun -np 10 ipython Helium_manuscript.py
 14%|█▍        | 144/1000 [01:00<05:52,  2.43it/s, Energy=65.49 ± 0.60 [σ²=1337.54, R̂=1.0034, τ=1.3<2.4]]
```

ps: to speed up the simulation, you can use multiple gpus/cpus with mpi.

**Comment by gpescia** [comment, marked as answer]
Hi @elad2718 ,

indeed the values for learning rate for SGD and diagonal shift for SR are not optimal it seems. I changed the learning rate to 0.005 and the diagonal shift to 0.001 and get an optimization curve as in the attachment. 
Does that help? 
[Helium_10_1d.log](https://github.com/netket/netket/files/10212302/Helium_10_1d.log)

**Reply by elad2718** [reply]
Thanks! I reproduced this on Colab. I consider this as an answer.

Not an expert, but I would call such strong dependence in meta parameters an issue in some sense. It could be great if you had a more robust algorithm for similar cases.

**Reply by gpescia** [reply]
I agree that the dependence on hyper parameters is an issue but I think it is a well known one in the machine-learning/optimization community. I think in this case the issue is a bit reinforced because of the divergent potential and the strong bias caused by the cusp condition. Sometimes this bias dominates and you get stuck at a local minimum like in this case (65 is the energy you get with only the cusp). 
I will update the example such that nobody runs into this again and maybe make a comment about the dependence. Thanks for raising this (:

**Reply by gpescia** [reply]
Maybe you could try to implement a potential as in https://journals.jps.jp/doi/epdf/10.7566/JPSJ.87.074002 Eq. 13 if you work with divergent potentials. Let me know if you need help with this.

**Reply by elad2718** [reply]
Looks interesting, but personaly I am intrested in applications to 3d few-body (3-20 particles) physics with strong (gaussian) interactions for nuclear physics calculations in the pionless EFT. 
I hope strong but not divergent interactions would be easier to work with.

Appreciate the swift and proffesional reply by both of you 👍
